planner/core/rule_inject_extra_projection.go

```diff
@@ -125,3 +131,70 @@ func injectProjBelowAgg(aggPlan PhysicalPlan, aggFuncs []*aggregation.AggFuncDes
 	aggPlan.SetChildren(proj)
 	return aggPlan
 }
+
+// The schema of PhysicalSort and PhysicalTopN are the same as the schema of
+// their children. When a projection is injected as the child of
+// PhysicalSort and PhysicalTopN, some extra columns will be added into the
+// schema of the Projection, thus we need to add another Projection upon
+// them to prune the redundant columns.
+func injectProjBelowSort(p PhysicalPlan, orderByItems []*ByItems) PhysicalPlan {
+	hasScalarFunc, numOrderByItems := false, len(orderByItems)
+	for i := 0; !hasScalarFunc && i < numOrderByItems; i++ {
+		_, isScalarFunc := orderByItems[i].Expr.(*expression.ScalarFunction)
+		hasScalarFunc = hasScalarFunc || isScalarFunc
+	}
+	if !hasScalarFunc {
+		return p
+	}
+
+	childPlan := p.Children()[0]
+	if childProj, isChildProj := childPlan.(*PhysicalProjection); isChildProj {
+		childProj.AvoidColumnEvaluator = true
```

Comment:
What does this mean? shall we also update the comments in `LogicalProjection` for `avoidColumnEvaluator`?

Reply:
Yes, we need to update the comment of LogicalProjection.

Reply:
Can you explain more on this flag? why its value for this projection is true, while is false for other 2 projections in this function?